tests/subsys/pm/power_mgmt/src/dummy_driver.c

```diff
@@ -9,33 +9,88 @@
 #include <pm/device_runtime.h>
 #include "dummy_driver.h"
 
+static bool dont_sleep;
+extern bool z_sys_post_kernel;
+
 static int dummy_open(const struct device *dev)
 {
-	return pm_device_get(dev);
+	dont_sleep = false;
+	pm_device_get(dev);
+	if (dev->pm->enable == false) {
+		pm_device_enable(dev);
+		return pm_device_get(dev);
+
+	}
+	return 0;
 }
 
 static int dummy_close(const struct device *dev)
 {
-	return pm_device_put(dev);
+	dont_sleep = false;
+	pm_device_put(dev);
+	if (z_sys_post_kernel == false) {
+		z_sys_post_kernel = true;
+		return 0;
+	}
+	pm_device_disable(dev);
+	return 0;
+}
+
+static int dummy_refuse_to_sleep(const struct device *dev)
+{
+	dont_sleep = true;
+	return 0;
+}
+
+static int dummy_not_wakeup_capable(const struct device *dev)
+{
+	atomic_clear_bit(&dev->pm->flags, PM_DEVICE_FLAGS_WS_CAPABLE);
+	return 0;
+}
+
+static int dummy_get_ready_to_close(const struct device *dev)
+{
+	dev->pm->state = PM_DEVICE_STATE_SUSPENDING;
+	return 0;
+}
+
+static int dummy_sys_pre_kernel(const struct device *dev)
+{
+	z_sys_post_kernel = false;
+	return 0;
 }
 
 static int dummy_device_pm_ctrl(const struct device *dev,
 				enum pm_device_action action)
 {
+	if (dont_sleep) {
+		if (action != PM_DEVICE_ACTION_RESUME) {
+			return -EIO;
+		}
+	}
```

Comment:
Please do not reinvent the wheel. `pm_device_wakeup_is_enabled(const struct device *dev)` exists exactly for this purpose

Reply:
when dont_sleep is set, return an error to prevent pm_system_suspend() from entering IDLE or SUSPEND.
if pm_device_wakeup_is_enabled() is true, this function has no chance to be called.